fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/MaterializedViewRewriter.java

```diff
@@ -866,6 +866,12 @@ protected OptExpression queryBasedRewrite(RewriteContext rewriteContext, ScalarO
                 Utils.compoundAnd(
                         rewriteContext.getQueryPredicateSplit().toScalarOperator(),
                         CompoundPredicateOperator.not(compensationPredicates)));
+        List<ScalarOperator> predicates = Utils.extractConjuncts(queryCompensationPredicate);
```

Comment:
`Utils.extractConjuncts`  is a little expensive, how about not ` Utils.compoundAnd` for `queryCompensationPredicate `.


```
 // query predicate and (not viewToQueryCompensationPredicate) is the final query compensation predicate
        ScalarOperator queryCompensationPredicate = MvUtils.canonizePredicate(
                Utils.compoundAnd(
                        rewriteContext.getQueryPredicateSplit().toScalarOperator(),
                        CompoundPredicateOperator.not(compensationPredicates)));
```

Reply:
the predicates should be removed after canonized. I think it is ok.